x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleTypeTests.java

```diff
@@ -817,6 +853,38 @@ public void testValidateFrozenPhaseHasSearchableSnapshot() {
         }
     }
 
+    public void testTimeseriesRegistryContainsAllActionsVersions() {
+        for (int version = INSTANCE.getLatestActionsOrderVersion(); version >= VERSION_ONE; version--) {
+            assertThat(ORDERED_VALID_HOT_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_WARM_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_COLD_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_FROZEN_ACTIONS.get(version), notNullValue());
+            assertThat(ORDERED_VALID_DELETE_ACTIONS.get(version), notNullValue());
+        }
+    }
+
+    public void testTimeseriesRegistryDoesntRemoveActionsInFutureVersions() {
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_HOT_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_WARM_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_COLD_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_FROZEN_ACTIONS);
+
+        assertHigerVersionsContainActionsInLowerVersions(ORDERED_VALID_DELETE_ACTIONS);
+    }
```

Comment:
These two tests encapsulate the versioning invariants: 
1. All versions must exist, from current all the way down to VERSION_ONE
2. All phases must contain all versions
3. Higher versions must contain all the actions in the previous version, irrespective of order

Reply:
> Higher versions must contain all the actions in the previous version, irrespective of order (i.e. we don't support removing actions)

Okay, I left a comment about this above, not sure if we should also have an assert for it in the code (though having a test does make me feel better)